### PR TITLE
replace stack driver log engine in Lucy with a log engine based on graphql polling

### PR DESCRIPTION
**Pull request by QuiteStochastic**
https://app.zenhub.com/workspaces/velocity-62264fddc441a100183f7f86/issues/gh/minaprotocol/mina/13259

### CODE REVIEW

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

File: src/lib/integration_test_cloud_engine/kubernetes_network.ml [hunk 1]
```diff
@@ -336,7 +336,7 @@ module Node = struct
                 ~metadata ;
               Deferred.Or_error.return result
           | Error (`Failed_request err_string) ->
-              [%log warn]
+              [%log debug]
```
Comment (mrmr1993) [inline src/lib/integration_test_cloud_engine/kubernetes_network.ml:339]: TODO: revert and handle this at the level of the annoying requests

File: src/lib/integration_test_cloud_engine/kubernetes_network.ml [hunk 2]
```diff
@@ -644,6 +660,54 @@ module Node = struct
...
     |> Deferred.bind ~f:Malleable_error.or_hard_error
 
+  let start_filtered_log ~logger ~log_filter t =
+    let open Deferred.Let_syntax in
+    [%log info] "Starting filtered log" ~metadata:(logger_metadata t) ;
+    let query_obj =
+      Graphql.StartFilteredLog.(make @@ makeVariables ~filter:log_filter ())
+    in
+    let%bind res =
+      exec_graphql_request ~logger ~node:t ~query_name:"StartFilteredLog"
```
Comment (mrmr1993) [inline src/lib/integration_test_cloud_engine/kubernetes_network.ml:670]: `~logger:(Logger.null ())` here

File: src/lib/integration_test_cloud_engine/kubernetes_network.ml [hunk 3]
```diff
@@ -644,6 +660,54 @@ module Node = struct
...
+  let get_filtered_log_entries ~logger ~last_log_index_seen t =
+    let open Deferred.Or_error.Let_syntax in
+    [%log info] "Getting logs from node, starting from log entry number %d"
+      last_log_index_seen ~metadata:(logger_metadata t) ;
+    let query_obj =
+      Graphql.GetFilteredLogEntries.(
+        make @@ makeVariables ~offset:last_log_index_seen ())
+    in
+    let%bind query_result_obj =
+      exec_graphql_request ~logger ~node:t ~query_name:"GetFilteredLogEntries"
```
Comment (mrmr1993) [inline src/lib/integration_test_cloud_engine/kubernetes_network.ml:702]: Et ici aussi

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]

**Pull request merged**